### PR TITLE
fix(build): Fix build failures due to dependency issue

### LINKED ISSUE #86: Failing Codeship build
Seems like the latest codeship build is failing on master according to the badge seen in README.

But I'm unable to investigate what's going wrong as [the codeship project is 404!](https://codeship.com/projects/714284d0-e914-0133-1e5d-4eaa3299b296)

Probably you're setting it to a kind of private mode?
Would it be possible for you to turn it public?

### PR DESCRIPTION
**Pull request by mumoshu**
TL;DR; revendor logrus 1.0.3 instead of 1.0.5

Fixes #86

---

Running `go test` for habitus within any clean go env was failing like this:

```
vendor/github.com/docker/docker/pkg/fileutils/fileutils.go:93: undefined: logrus.Debugf
vendor/github.com/docker/docker/pkg/fileutils/fileutils_unix.go:17: undefined: logrus.Errorf
vendor/github.com/docker/docker/pkg/mount/mount.go:83: undefined: logrus.Debugf
vendor/github.com/docker/docker/pkg/mount/mount.go:103: undefined: logrus.WithError
vendor/github.com/docker/docker/pkg/mount/mount.go:107: undefined: logrus.Debugf
FAILgithub.com/cloud66-oss/habitus [build failed]
```

It was due to that we had been vendoring an incorrect version of logrus(1.0.5). logrus is a transitive dependency of habitus, that is used only from within the `docker/docker` pkg. `docker/docker` doesn't use `dep` and hence doesn't include `Gopkg.lock`, which resulted in `dep` to determine which version of logrus to vendor. I updated `Gopkg.lock` and then ran `dep ensure -vendor-only` to update the vendored logrus and the issue has gone.

---

Side note:

I intentionally modified `Gopkg.lock` manually.

Normally, it would be a matter of updating `Gopkg.toml` and running `dep ensure` variants to update the vendored logrus. But there seem to be an issue that `dep` doesn't take the version number for logrus in our `Gopkg.toml`. That is, running `dep ensure` doesn't update `Gopkg.lock` with logrus 1.0.3, even when I followed [the official guidance](https://github.com/golang/dep/blob/master/docs/FAQ.md#how-do-i-constrain-a-transitive-dependency-s-version).

What I tried is: (1) add `required = ["github.com/sirupsen/logrus"]` and (2) use either `[[constraint]]` or `[[override]]` to stick the version number and (2) run `dep ensure` or `dep ensure -update`. So please feel free to reproduce and "fix" the Gopkg.toml so that `dep ensure` alone just works.

### CODE REVIEW

**Comment by mumoshu** [comment]
Please don't squash, but merge these 3 commits as-is instead. I squashed these on my local machine, and it didn't fix the sirupsen-vs-Sirupsen issue I noted in 706b966 and 0fdfd15.
